Related process

bdmbdsm opened the pull request.

Commit:
RelatedProcesses

- Fix code style issues
- Add pass statement to a blank class
- Forbid asset owner to create & edit relatedProcesses
- Use migration in the includeme
- Update migration logic - make it independent from web-related stuff
- Add `get_snapshot` tool
- Bounce asset snapshot
- Use snapshot mechanism
- Add status check to the migration
- Require related lot to activate asset
- Set related process type on creation in the tests
- Remove `relatedLot` attr from the asset model

leits commented on openregistry/assets/bounce/migration.py:
```diff
@@ -0,0 +1,74 @@
+            return
+
+        related_lot_id = asset.get('relatedLot')
+        if related_lot_id is None:
```
`relatedLot` не буде присутній у всіх статусах окрім `active` i `complete`: http://assetsbounce.api-docs.registry.ea2.openprocurement.io/en/latest/assetsWorkflow.html

Потрібно добавити перевірку на статус у `_skip_predicate`

bdmbdsm replied:
Додав. Чомусь тут не відображається. Ось [коміт](https://github.com/openprocurement/openregistry.assets.bounce/pull/45/commits/74ec1474deb400938b93f7da7a8167b8789902bf).

leits replied:
Якщо тут немає, значить загальний патч змін не принесе ці зміни у кінцеву гілку

leits replied:
Але вони є, все ок :+1:

oleksiyVeretiuk commented on openregistry/assets/bounce/models.py:
```diff
@@ -112,6 +116,6 @@ def validate_documents(self, data, docs):
-        if data['status'] == 'active' and not lot:
-            raise ValidationError(u'This field is required.')
+    def validate_relatedProcesses(self, context, related_processes):
+        if context['status'] == 'active' and not related_processes:
```
Чи не краще, зробити перевірку на relatedProcess, який має в собі type Lot, адже по великому рахунку важливе не наявність relatedProcess-ів, а наявність relatedProcess-a, який прив'язаний до лота

bdmbdsm replied:
Правильно, - виправлю.

oleksiyVeretiuk commented on openregistry/assets/bounce/models.py:
```diff
@@ -74,16 +72,22 @@ class Asset(BaseAsset):
+    # All documents and attachments related to the asset
+    documents = ListType(ModelType(AssetDocument), default=list())
+    relatedProcesses = ListType(ModelType(RelatedProcess), default=list(), max_size=1)
+    relatedLot = StringType()
```
relatedLot заміняється relatedProcesses, чи не треба його звідси прибрати

bdmbdsm replied:
Згоден. Приберу.

Commit:
Generate docs with relatedProcesses within

- Rename `docs.py` to `docs/update_tutorial_requests.py`
- Make `update_tutorial_requests` sctipt work
- Update tutorial doc's source request files
- Provide documentation build manual in the readme
- Fix code highlighting in the readme
- Run `gen-requests` script on Travis

Commit:
Remove add_related_processes_views call

Commit:
Fix update_tutorial_requests test
